src/platform/site-wide/cta-widget/index.js

```diff
@@ -35,14 +41,17 @@ const MHV_ACCOUNT_TYPES = ['Premium', 'Advanced', 'Basic'];
 export class CallToActionWidget extends React.Component {
   constructor(props) {
     super(props);
+    const { appId } = props;
+    this._isHealthTool = HEALTH_TOOLS.includes(appId);
     this._popup = null;
+    this._continueUrl = continueUrl(appId);
     this._redirectUrl = redirectUrl(window.location.pathname);
-    this._requiredServices = requiredServices(props.appId);
-    this._serviceDescription = serviceDescription(props.appId);
+    this._requiredServices = requiredServices(appId);
+    this._serviceDescription = serviceDescription(appId);
   }
 
   componentDidMount() {
-    if (this.isHealthTool() && this.props.isLoggedIn) {
+    if (this._isHealthTool && this.props.isLoggedIn) {
```

Comment:
Yeah, I doubt this gets touched because every page load will start a request for profile. I was just trying to be safe in case the request is completed before this.

I'm often unsure whether we should rely on `isLoggedIn` or the session token, and the former makes me feel iffy about whether we should account for it being available by the time the component is mounted due to the async nature of fetching it. To that end, I try to account for it just in case.

Reply:
Ah yeah, I've run into the same predicament a couple times. That makes sense then!

Reply:
FYI, session token will eventually go away. Please don't rely on it in anywhere other than sign-in.

The plan is to replace session token with a cookie, presence of that cookie will allow you to make authenticated risks. You'll no longer need session token and authentication headers.